src/python/WMCore/WMBS/MySQL/Jobs/AddWorkUnits.py

```diff
@@ -55,6 +55,8 @@ def execute(self, jobid=None, fileid=None, run=None, lumi=None, jobFileRunLumis=
             binds = self.getBinds(jobFileRunLumis)
         elif jobid and fileid and run and lumi:
             binds = DBFormatter.getBinds(self, jobid=jobid, fileid=fileid, run=run, lumi=lumi)
+        elif not jobFileRunLumis and isinstance(jobFileRunLumis, list):  # Empty list is ok
```

Comment:
Don't we reach the same result with the `isinstance` check only (since the if above takes care of element'ed list)?

Reply:
No. If jobFileRunLumis is None, this line will not be true. Only if its an empty list.

Reply:
That's my point, at this stage the only thing you care about is the data type. None or [] has the same outcome in the first statement (here comes my suggestion to keep only the second one)

Reply:
It's an and not an or. If all the parameters are None, it falls through to the end and generates an error message.

Reply:
@ericvaandering, I think Alan is right it seems it give the same result.
But I am not sure why this is needed, do you want to avoid error message printing out?
Is there case this calls with everything is None. If we don't expect that we should raise an error. Not just printing out the log message.